mailscanner/password_reset.php

```diff
@@ -0,0 +1,429 @@
+        $uid = $link->real_escape_string($_POST['uid']);
+        if ($_POST['pwd1'] === $_POST['pwd2']) {
+            //passwords match, now we need to store them
+            //first, check form hasn't been modified
+            $sql = "SELECT resetid FROM users WHERE username = '$email'";
+            $result = dbquery($sql);
+            if (!$result) {
+                die("Error: ".$link->error);
+            }
+            $row = $result->fetch_array();
+            if ($row['resetid'] == $_POST['uid']) {
+                require_once(MAILWATCH_HOME . '/lib/password.php');
+                $password = $link->real_escape_string(password_hash($_POST['pwd1'], PASSWORD_DEFAULT));
+                $lastreset = time();
+                $sql = "UPDATE users SET password = '$password', resetid = '', resetexpire = '0', lastreset ='$lastreset' WHERE username ='$email'";
+                $result = dbquery($sql);
+                if (!$result) {
+                    die(__('errorpwdchange100'));
+                }
+
+                //now send email telling user password has been updated.
+                $html = '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
+<html>
+<head>
+ <title>'.__('01pwdresetemail100').'</title>
+ <style type="text/css">
+ <!--
+  body, td, tr {
+  font-family: sans-serif;
+  font-size: 8pt;
+ }
+ -->
+ </style>
+</head>
+<body style="margin: 5px;">
+
+<!-- Outer table -->
+<table width="100%%" border="0">
+ <tr>
+  <td><img src="'. IMAGES_DIR . MW_LOGO .'" alt="'.__('mwlogo99').'"/></td>
+  <td align="center" valign="middle">
+   <h2>'.__('02pwdresetemail100').'</h2>
+   <p>'.sprintf(__('03pwdresetemail100'), $email) . '</p>
+  </td>
+ </tr>
+ </table>
+</body>
+</html>';
+                $text = sprintf(__('04pwdresetemail100'), $email);
+
+                //Send email
+                $mime = new Mail_mime("\n");
+                if (defined('PWD_RESET_FROM_NAME') && defined('PWD_RESET_FROM_ADDRESS') && PWD_RESET_FROM_NAME !== '' && PWD_RESET_FROM_ADDRESS !== '') {
+                    $sender = PWD_RESET_FROM_NAME . '<' . PWD_RESET_FROM_ADDRESS . '>';
+                } else {
+                    $sender = QUARANTINE_REPORT_FROM_NAME . ' <' . QUARANTINE_FROM_ADDR . '>';
+                }
+                $hdrs = array(
+                    'From' => $sender,
+                    'To' => $email,
+                    'Subject' => __('02emailsubject100'),
+                    'Date' => date("r")
+                );
+                $mime->addHTMLImage(MAILWATCH_HOME . IMAGES_DIR . MW_LOGO, 'image/png', MW_LOGO, true);
+                $mime->setTXTBody($text);
+                $mime->setHTMLBody($html);
+                $body = $mime->get();
+                $hdrs = $mime->headers($hdrs);
+                $mail_param = array('host' => QUARANTINE_MAIL_HOST);
+                $mail =& Mail::factory('smtp', $mail_param);
```

Comment:
it's better to bypass Mail factory and use a new instance of Mail_smtp, like done on quarantine_report.php cron script